Add testing and base devel images

SantiagoTorres opened the pull request:
I reorganized the repo a little bit and added two candidate new images to host.

Let me know if this is what you had in mind.

bastelfreak commented on testing/pacman.conf:
```diff
@@ -0,0 +1,100 @@
+[extra]
+Include = /etc/pacman.d/mirrorlist
+
+#[community-testing]
```
shouldn't we include this as well?

SantiagoTorres replied:
Right,  I wasn't sure to whether enable this, or have yet-another-image™ called community-testing. Either approaches sound good to me.

bastelfreak replied:
Depends on the usecase I guess. For the stuff I would like to use the images I prefer an image with all testing repos.

SantiagoTorres replied:
Sounds reasonable, let me fix this up 👍

polyzen replied:
> If you enable testing, you must also enable community-testing. If you enable any other testing repository listed in the following subsections, you must also enable testing.

SantiagoTorres commented:
I updated this PR in a more reasonable way:

- I re-use the dockerfile/rootfs/etc. from the base image, as well as the Makefile,  which now loads all resources locally
- The testing pacman.conf is the same as the one in /usr/share/devtools/pacman-testing.conf
- Avoided the mass-renaming and other changes that polluted the diff quite a lot
- Updated the README

I'm still curious as to how to update the Makefile to test all the images. I'm not even sure if that's necessary. What do you guys think? 

@bastelfreak what do you think of this approach?

SantiagoTorres commented on Makefile:
```diff
@@ -31,4 +31,4 @@ docker-push: docker-image-test
 	docker push $(DOCKER_ORGANIZATION)/$(DOCKER_IMAGE)
 
-.PHONY: rootfs docker-image docker-image-test ci-test docker-push
+.PHONY: rootfs docker-image docker-image-test ci-test docker-push base
```
I'll remove this phony target once review is done. Sorry about this.

pierres commented:
Wouldn't it be more Docker like if we just create a Dockerfile that extends the base image and just enables the testing repo or adds more packages? I#d say its best if we derive everything from a single base image.
